bionic/core/flow_execution.py

```diff
@@ -66,9 +66,26 @@ def run(self, states):
                 # their cache entries deleted, their status will be correctly updated
                 # to "not complete".
                 for dep_state in entry.state.dep_states:
-                    self._get_or_create_entry_for_state(dep_state)
+                    dep_entry = self._get_or_create_entry_for_state(dep_state)
```

Comment:
As we discussed on Slack, WDYT about replacing `is_requested` with `is_needed_in_memory` and having that be an argument to `_get_or_create_entry_for_state`?

Reply:
While making this change, I realized that I'm setting up caching flags for state inside `_get_or_create_entry_for_state`. We need the should persist cache flag to determine whether an entry is needed in memory.

Wdyt about passing `is_requested` to  `_get_or_create_entry_for_state` but replacing `is_requested` with `is_needed_in_memory` inside the entry?

Reply:
This is going to be easier to address in #205, so I'm going to address it in that PR.